dev/CommonStyles/TestUI/CompactPage.xaml.cs

```diff
@@ -0,0 +1,104 @@
+using System;
+
+namespace MUXControlsTestApp
+{
+    public sealed partial class CompactPage : TestPage
+    {
+        ObservableCollection<FontFamily> fonts = new ObservableCollection<FontFamily>();
+
+        public CompactPage()
+        {
+            this.InitializeComponent();
+            fonts.Add(new FontFamily("Arial"));
+            fonts.Add(new FontFamily("Courier New"));
+            fonts.Add(new FontFamily("Times New Roman"));
+        }
+
+        private void AutoSuggestBox_TextChanged(AutoSuggestBox sender, AutoSuggestBoxTextChangedEventArgs args)
+        {
+            if (args.Reason == AutoSuggestionBoxTextChangeReason.UserInput)
+            {
+                var suggestList = new List<string>();
+                suggestList.Add("001");
+                suggestList.Add("002");
+                suggestList.Add("003");
+                suggestList.Add("004");
+                suggestList.Add("005");
+
+                sender.ItemsSource = suggestList.Select(text => sender.Text + text);
+            }
+        }
+
+        private void VerifyHeight(SimpleVerify simpleVerify, FrameworkElement item, int expectHeight, string controlName)
+        {
+            if (item != null)
+            {
+                // Avoid Height because of CornerRadius
+                if (Math.Abs(item.ActualHeight - expectHeight) > 0.5)
+                {
+                    simpleVerify.IsEqual(item.ActualHeight.ToString(), expectHeight, controlName + ".ActualHeight=" + expectHeight);
+                }
+            }
+            else
+            {
+                simpleVerify.IsTrue(false, "Can't find " + controlName);
+            }
+        }
+
+        private void VerifyChildHeight(SimpleVerify simpleVerify, FrameworkElement item, string childXName, int expectHeight, string controlName)
+        {
+            FrameworkElement child = GetChild(item, childXName);
+            VerifyHeight(simpleVerify, child, expectHeight, controlName + "." + childXName);
+        }
+
+        private FrameworkElement GetChild(FrameworkElement item, string childXName)
+        {
+            FrameworkElement root = (item == null) ? null: (FrameworkElement)VisualTreeHelper.GetChild(item, 0);
+            FrameworkElement child = (root == null) ? null: (FrameworkElement)root.FindName(childXName);
+            return child;
+        }
+        private void RunTest_Click(object sender, RoutedEventArgs e)
+        {           
+            SimpleVerify simpleVerify = new SimpleVerify();
+            FrameworkElement item = ListView.FindElementOfTypeInSubtree<ListViewItem>();
+            VerifyHeight(simpleVerify, item, 32, "ListViewItem");
+            item = TreeView.FindElementOfTypeInSubtree<TreeViewItem>();
+            VerifyHeight(simpleVerify, item, 24, "TreeViewItem");
+
+            VerifyHeight(simpleVerify, NavItem1, 32, "NavigationViewItem");
+
+            VerifyChildHeight(simpleVerify, TextBox, "BorderElement", 24, "TextBox");
```

Comment:
Could use nameof(TextBox) on all these to make sure they're always exact matches

Reply:
TextBox includes Header, so we can't do it by TextBox.Height directly.

Reply:
The goal of the string is to match the named member, right?

Reply:
yes